### PR TITLE
Remove visitor ordering and order selection.

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR

### CODE REVIEW

**Review by vekterli — APPROVED** [review]
:+1: 🎉a couple of comments/question for consideration

File: document/src/main/java/com/yahoo/document/select/rule/IdNode.java [hunk 1]
```diff
@@ -88,13 +82,8 @@ public Object evaluate(Context context) {
...
             throw new IllegalStateException("Document id doesn't have doc type.");
         } else if (field.equalsIgnoreCase("order")) {
-            if (id.getScheme() instanceof OrderDocIdString) {
-                OrderDocIdString ods = (OrderDocIdString)id.getScheme();
-                if (ods.getWidthBits() == widthBits && ods.getDivisionBits() == divisionBits) {
-                    return ods.getOrdering();
-                }
-            }
-        } else{
+            // TODO Remove from grammar on VESPA 8
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply document/src/main/java/com/yahoo/document/select/rule/IdNode.java]: Fixed

File: documentapi/src/main/java/com/yahoo/documentapi/messagebus/protocol/RoutableFactories60.java [hunk 2]
```diff
@@ -229,7 +228,7 @@ protected boolean doEncode(DocumentMessage obj, DocumentSerializer buf) {
                 buf.put(null, b);
             }
 
-            buf.putInt(null, msg.getVisitorOrdering());
+            buf.putInt(null, 0);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply documentapi/src/main/java/com/yahoo/documentapi/messagebus/protocol/RoutableFactories60.java]: Fixed

File: document/src/test/java/com/yahoo/document/GlobalIdTestCase.java [hunk 3]
```diff
@@ -86,10 +86,10 @@ private void verifyGidToBucketIdMapping(String idString) {
 
     @Test
     public void testToBucketId() {
-        verifyGidToBucketIdMapping("userdoc:ns:1:abc");
-        verifyGidToBucketIdMapping("userdoc:ns:1000:abc");
-        verifyGidToBucketIdMapping("userdoc:hsgf:18446744073700000000:dfdfsdfg");
-        verifyGidToBucketIdMapping("groupdoc:ns:somegroup:hmm");
+        verifyGidToBucketIdMapping("id:ns:mytype:n=1:abc");
+        verifyGidToBucketIdMapping("id:ns:mytype:n=1000:abc");
+        verifyGidToBucketIdMapping("id:hsgf:mytype:n=9146744073700000000:dfdfsdfg");
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply document/src/test/java/com/yahoo/document/GlobalIdTestCase.java]: Current limitations is that it must be parseable as Long.

File: storage/src/vespa/storage/distributor/operations/external/visitoroperation.cpp [hunk 4]
```diff
@@ -649,57 +567,8 @@ VisitorOperation::pickTargetNode(
 }
 
 bool
-VisitorOperation::documentSelectionMayHaveOrdering() const
+VisitorOperation::parseDocumentSelection(DistributorMessageSender& )
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply storage/src/vespa/storage/distributor/operations/external/visitoroperation.cpp]: Removed

**Pull request merged**